rabbit-image/README.md

```diff
@@ -0,0 +1,188 @@
+# RabbitMQ Cluster Image
+This docker image definition provides a RabbitMQ Server cluster for use in Kubernetes.  It is designed to be used in a
+Kubernetes [StatefulSet](https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/) which will 
+persist the backing "Mnesia" filestore and provide consistent pod names for clustering.  This image and it's associated
+deployment are designed to be fault-tolerant.  If a single node goes down, there will be no service interruption while
+Kubernetes stands the replacement node back up.
+
+NOTE that this image is also configured to provide [High-Availability mirrored Queues](https://www.rabbitmq.com/ha.html)
+with all queues being mirrored across 2 nodes by default.
+
+NOTE that this image is using the [RabbitMQ Autocluster plugin](https://github.com/rabbitmq/rabbitmq-autocluster) to
+provide clustering with a version built off of commit 1e7f9710e8802ef51d23692c344f64b051a8c331 due to the latest release
+missing some bug fixes and being bound to an old version of RabbitMQ.
+
+## Details
+A deployment environment will generally want at least 3 RabbitMQ instances running.  We will also need an ETCD service 
+running to provide cluster coordination.
+
+The deployment should not require direct intervention.
+
+## Usage Example
+Sample Kubernetes YAML:
+```yaml
+# Create a shared secret used by rabbit-mq cluster nodes to authenticate with each other
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rabbit-secret
+type: Opaque
+data:
+  erlang-cookie: ZXhhbXBsZSBzZWNyZXQ=
+---
+
+# Create a service wrapping the Rabbit ETCD pod
+apiVersion: v1
+kind: Service
+metadata:
+  name: rabbit-etcd
+spec:
+  type: ClusterIP
+  selector:
+    app: rabbit-etcd
+  ports:
+  - name: server-port
+    port: 2379
+    targetPort: 2379
+  - name: client-port
+    port: 4001
+    targetPort: 4001
+---
+
+# Create an ETCD deployment for coordinating the cluster
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: rabbit-etcd-deployment
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: rabbit-etcd
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+      - name: etcd-rabbit
+        image: microbox/etcd:2.1.1
```

Comment:
I really hate to stand up another key-value store when we already have Redis running, but I don't have time to extend the autocluster plugin with redis functionality and don't want to rewrite the RabbitMQ cluster coordination logic in bash.